### PR TITLE
Update docs for transform plugins

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Update comments for `Correct` and `AstyneAnnotations` of `transform`
- Update documentation for `transform` plugins

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [X] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [X] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/source/docs/command-reference/context_free/transform.md [hunk 1]
```diff
@@ -101,7 +101,9 @@ Basic dataset item manipulations:
 - [`remove_images`](#remove_images) - Removes specific images
 - [`remove_annotations`](#remove_annotations) - Removes annotations
 - [`remove_attributes`](#remove_attributes) - Removes attributes
-- [`astype_annotations`](#astype_annotations) - Convert annotation type
+- [`correct`](#correct) - Corrects annotaiton types
+- [`astype_annotations`](#astype_annotations) - Transforms annotation types
+- [`clean`](#clean) - Remove noisy data for tabular dataset
```
Comment (jihyeonyi) [inline docs/source/docs/command-reference/context_free/transform.md]: Remove -> Removes

File: src/datumaro/plugins/transforms.py [hunk 2]
```diff
@@ -1746,13 +1758,15 @@ def __iter__(self):
...
-    Based on a specified mapping, it transforms the annotation types,|n
-    changing them to 'Label' if they are categorical, and to 'Caption' if they are of type string, float, or integer.|n
+    This transform changes annotations to 'Label' if they are categorical, and to 'Caption'
+    if they are of type string, float, or integer. This is particularly useful when working
+    with tabular data that needs to be converted into a format suitable for specific machine
+    learning tasks.|n
     |n
     Examples:|n
-        - Convert type of `title` annotation|n
+        - Convert the type of a `title` annotation:|n
```
Comment (jihyeonyi) [inline src/datumaro/plugins/transforms.py]: Convert -> Converts

File: docs/source/docs/command-reference/context_free/transform.md (hunk 1 above)
Comment (sooahleex) [reply docs/source/docs/command-reference/context_free/transform.md]: Fix in [c79dae8](https://github.com/openvinotoolkit/datumaro/pull/1599/commits/c79dae8f38c555150d38887fcd08379ed5a0b5c2)

File: src/datumaro/plugins/transforms.py (hunk 2 above)
Comment (sooahleex) [reply src/datumaro/plugins/transforms.py]: Fix in [c79dae8](https://github.com/openvinotoolkit/datumaro/pull/1599/commits/c79dae8f38c555150d38887fcd08379ed5a0b5c2)

**Review by jihyeonyi — APPROVED** [review]

**Pull request merged**